### PR TITLE
CI for TIM-VX backend

### PR DESCRIPTION
**Pull request by fengyuentau**
Related PR: https://github.com/opencv/opencv/pull/21036
Docker image used for x86_64 simulator: https://hub.docker.com/r/yuentau/ocv_ubuntu
A running sample of this workflow on my fork: https://github.com/fengyuentau/opencv/actions/runs/2003773534

The yaml file must be on the **default** branch for GitHub to pick up and run the workflow. I propose to add the condition if for the two jobs before the PR for TIM-VX backend is merged. In this case, a new label "category:dnn_timvx" needs to be created for pull requests.

Also we need to add the Khadas VIM3 as the self-hosted runner to run the second job, which requires some operations by the admin.

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: .github/workflows/extra_tests.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  pull_request:
+    branches: [ 4.x ]
+    types: [ labeled, opened, synchronize, reopened ]
+
+
+jobs:
+  x86-simulator-build-test:
+    if: contains(github.event.pull_request.labels.*.name, 'category:dnn_timvx')
+    runs-on: ubuntu-20.04
+    container: yuentau/ocv_ubuntu:20.04
```
Comment (asmorkalov) [inline .github/workflows/extra_tests.yml:13]: Please add reference to Dockerfile as comment at least.

File: .github/workflows/extra_tests.yml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    env:
+      PR_AUTHOR: ${{ github.event.pull_request.user.login }}
+      SOURCE_BRANCH_NAME: ${{ github.head_ref }}
+      TARGET_BRANCH_NAME: ${{ github.base_ref }}
+    steps:
+      - name: info
+        run: |
+          echo "PR Author: ${{ env.PR_AUTHOR }}""
```
Comment (asmorkalov) [inline .github/workflows/extra_tests.yml:21]: Extra `"`

File: .github/workflows/extra_tests.yml [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+      - name: merge opencv with test branch
+        shell: bash
+        run: |
+          cd opencv
+          git config user.email "opencv.ci"
+          git config user.name "opencv.ci"
+          git pull -v "https://github.com/${{ env.PR_AUTHOR }}/opencv" "${{ env.SOURCE_BRANCH_NAME }}" --allow-unrelated-histories
+      - name: configure
+        run: |
+          cmake -B build -DWITH_TIMVX=ON -DCMAKE_INSTALL_PREFIX=./install -DBUILD_SHARED_LIBS=ON -DBUILD_PERF_TESTS=OFF -DBUILD_EXAMPLES=OFF -DBUILD_DOCS=OFF -DWITH_OPENCL=OFF -DBUILD_TESTS=OFF opencv
```
Comment (asmorkalov) [inline .github/workflows/extra_tests.yml:43]: I propose to enable tests for CI.

File: .github/workflows/extra_tests.yml [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+      cancel-in-progress: false
+    runs-on: [self-hosted, Linux, ARM64, khadas-vim3]
+    env:
+      PR_AUTHOR: ${{ github.event.pull_request.user.login }}
+      SOURCE_BRANCH_NAME: ${{ github.head_ref }}
+      TARGET_BRANCH_NAME: ${{ github.base_ref }}
+    steps:
+      - name: info
+        run: |
+          echo "PR Author: ${{ env.PR_AUTHOR }}""
```
Comment (asmorkalov) [inline .github/workflows/extra_tests.yml:60]: Extra `"`

File: .github/workflows/extra_tests.yml (hunk 1 above)
Comment (fengyuentau) [reply .github/workflows/extra_tests.yml]: Done.

File: .github/workflows/extra_tests.yml (hunk 2 above)
Comment (fengyuentau) [reply .github/workflows/extra_tests.yml]: Removed.

File: .github/workflows/extra_tests.yml (hunk 3 above)
Comment (fengyuentau) [reply .github/workflows/extra_tests.yml]: Set BUILD_TESTS=ON at the latest commit.

File: .github/workflows/extra_tests.yml [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+            echo "no merge since ${{ env.PR_AUTHOR }}/opencv_extra does not have branch ${{ env.SOURCE_BRANCH_NAME }}"
+          fi
+      - name: configure
+        run: |
+          cmake -B build -D CMAKE_BUILD_TYPE=RELEASE -DCMAKE_INSTALL_PREFIX=./install -DWITH_TIMVX=ON -DWITH_OPENCL=OFF -DWITH_EIGEN=OFF opencv
+      - name: build
+        run: cmake --build build --target opencv_test_dnn -j 4
+      - name: unit tests for int8 layers
+        run: |
+          OPENCV_TEST_DATA_PATH=./opencv_extra/testdata ./build/bin/opencv_test_dnn --gtest_filter="Test_Int8_layers.*/1"
```
[the remark this answers is not in the document]
Comment (fengyuentau) [reply .github/workflows/extra_tests.yml]: `/1` for only tests on NPU to save time running the whole pipeline on the board.

File: .github/workflows/extra_tests.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (fengyuentau) [reply .github/workflows/extra_tests.yml]: Reference added. Dockerfile can be accessed in the overview section of the linked Docker Hub repo.

File: .github/workflows/extra_tests.yml [hunk 6]
```diff
@@ -0,0 +1,104 @@
+name: TIM-VX Backend
```
[the remark this answers is not in the document]
Comment (fengyuentau) [reply .github/workflows/extra_tests.yml]: Renamed to "timvx_backend_tests.yml". Or any suggestions?

File: .github/workflows/extra_tests.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (fengyuentau) [reply .github/workflows/extra_tests.yml]: I make the source more clear: `container: docker.io/yuentau/ocv_ubuntu:20.04`. It works as the same with the one without `docker.io/` prefix. You can verify by checking 'Initialize containers' from my test https://github.com/fengyuentau/github-actions-tests/runs/5653954490?check_suite_focus=true

**Review by vpisarev — APPROVED** [review]

**Review by asmorkalov — APPROVED** [review]

**Pull request merged**